### ISSUE TITLE
[@typescript-eslint/naming-convention] @typescript-eslint/naming-convention should disable `camelcase`

### ISSUE THREAD

**Issue by Mister-Hope** [typescript-eslint/typescript-eslint#2062]
The `@typescript-eslint/camelcase` is changed to `@typescript-eslint/naming-convention` in V3, that's cool, but `@typescript-eslint/camelcase` disable `camelcase` for ts files in V2, so should the new `@typescript-eslint/naming-convention` in V3.

Now, file like this will throw camelcase error:

```ts
/* eslint-disable @typescript-eslint/naming-convention */

  md.renderer.rules.footnote_ref = renderFootnoteRef;
  md.renderer.rules.footnote_block_open = renderFootnoteBlockOpen;
  md.renderer.rules.footnote_block_close = renderFootnoteBlockClose;
  md.renderer.rules.footnote_open = renderFootnoteOpen;
  md.renderer.rules.footnote_close = renderFootnoteClose;
  md.renderer.rules.footnote_anchor = renderFootnoteAnchor;

  // helpers (only used in other rules, no tokens are attached to those)
  md.renderer.rules.footnote_caption = renderFootnoteCaption;
  md.renderer.rules.footnote_anchor_name = renderFootnoteAnchorName;
```

```
Z:\npm\vuepress-theme-hope\packages\md-enhance\src\markdown-it\footnote.ts
  110:21  error  Identifier 'footnote_ref' is not in camel case          camelcase
  111:21  error  Identifier 'footnote_block_open' is not in camel case   camelcase
  112:21  error  Identifier 'footnote_block_close' is not in camel case  camelcase
  113:21  error  Identifier 'footnote_open' is not in camel case         camelcase
  114:21  error  Identifier 'footnote_close' is not in camel case        camelcase
  115:21  error  Identifier 'footnote_anchor' is not in camel case       camelcase
  118:21  error  Identifier 'footnote_caption' is not in camel case      camelcase
  119:21  error  Identifier 'footnote_anchor_name' is not in camel case  camelcase
```

**Comment by Mister-Hope** [comment]
In the previous V2, typescript disable `camelcase` rule in eslint for me, and it works fine if I use `/* eslint-disable @typescript-eslint/camelcase */`.

In V3，`camelcase` will not disabled so I have to add `/* eslint-diable camelcase */` to prevent eslint throwing errors. 

Also, if I add `/* eslint-disable @typescript-eslint/camelcase */`, I will case this error:

```
error  Definition for rule '@typescript-eslint/camelcase' was not found  @typescript-eslint/camelcase
```

If I remove it, I will get issue in my IDE problems panel:

![image](https://user-images.githubusercontent.com/33315834/82637538-9549d680-9c37-11ea-9b0e-d52ee8b3ecc6.png)

**Comment by bradzacher** [comment]
In v2, we had our `camelcase` rule recommended, so that's why it disabled the base `camelcase` rule.

In v3 we've removed all naming rules from our recommended set due to the number of users that disagreed with the stylistic choice being in recommended.
Because it's not recommended, we can't automatically disable the rule.

Unfortunately eslint provides no mechanism for one rule to influence the config of another rule.
Every rule is run in isolation, and configuration is locked in ahead of time.

I've updated the rule's docs to suggest disabling `camelcase`.
